src/main/java/io/confluent/idesidecar/restapi/messageviewer/strategy/ConfluentCloudConsumeStrategy.java

```diff
@@ -222,6 +225,34 @@ private PartitionConsumeData processPartition(
     );
   }
 
+  private static List<SimpleConsumeMultiPartitionResponse.PartitionConsumeRecordHeader>
+  decodeHeaders(PartitionConsumeRecord record) {
+    return record
+        .headers()
+        .stream()
+        .map(header -> {
+          String decodedValue;
+          try {
+            decodedValue = new String(
+                BASE64_DECODER.decode(header.value()), StandardCharsets.UTF_8
+            );
+          } catch (IllegalArgumentException e) {
+            // For whatever reason, we couldn't decode the Base64 encoded header value.
+            // Perhaps the API changed or the data is corrupted. Either way,
+            // we log a warning and return the raw value.
+            Log.warn("Failed to base64 decode header value from Confluent Cloud", e);
```

Comment:
I'm worried about the sidecar spewing this log message w/ stack trace for a lot of messages that all fit the same pattern. In other words, **if** this happens for one record, it's probably going to spew for every record being consumed.

Is it straightforward to count these _per request_, and maybe log it only for the first one (or few)?

Reply:
Hm true, I think the "only log for first one" will be easy to implement once we have #153 in place.

Reply:
I'm starting to think that this is probably not a failure case after all. The records may have header values that are not UTF-8 strings.

Reply:
> I'm starting to think that this is probably not a failure case after all. The records may have header values that are not UTF-8 strings.

What about changing the log level to debug?

Reply:
Makes sense. Changed log level to debug and added more info to message.